### ISSUE TITLE
Incorrect nullability warnings for new interpolated string handlers

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#57528]
**Version Used**: 
```
Branch main (30 Oct 2021)
Latest commit 63ee098 by Fred Silberberg:
Forbid interpolated string handler conversions that reference an instance as indexer arguments in nested member initializers (#57456)

Followup to LDM discussion. This scenario is pathological, and we don't want to support it.
```

**Steps to Reproduce**:

Compile the following code:
```cs
using System.Runtime.CompilerServices;
using System.Text;

#nullable enable

M(out string? s1, $"");

void M(out string? t1, [InterpolatedStringHandlerArgument("t1")] CustomHandler c) => t1 = null;

[System.Runtime.CompilerServices.InterpolatedStringHandler]
public partial struct CustomHandler
{
  public CustomHandler(int literalLength, int formattedCount, out string? t1, out bool success) : this()
  {
    t1 = default;
    success = true;
  }
}

public partial struct CustomHandler
{
  private readonly StringBuilder _builder = new();
  public bool AppendLiteral(string literal) => true;
  public bool AppendFormatted(object o, int alignment = 0, string? format = null) => true;
}
```

**Expected Behavior**:
No warnings

**Actual Behavior**:
`CS8601: Possible null reference assignment.` is reported for `out string? t1`.
`t1` is already declared as `string?` so nullable assignments to it shouldn't warn. The warning also cannot be suppressed easily as suppressions aren't allowed on declaration expressions

**Comment by 333fred** [comment]
Also closing this as a duplicate of https://github.com/dotnet/roslyn/issues/54583.
